crates/criticaltrust/src/signatures/keychain.rs

```diff
@@ -32,6 +36,37 @@ impl Keychain {
         Ok(keychain)
     }
 
+    pub fn keys(&self) -> &HashMap<KeyId, PublicKey> {
+        &self.keys
+    }
+
+    pub fn revocation_info(&self) -> Option<RevocationInfo> {
```

Comment:
I was able to make this

```rust
    pub fn revocation_info(&self) -> Option<&RevocationInfo> {
        self.revocation_info.as_ref()
    }
```

Without other changes. Is there a reason to force the clone?

Reply:
Oh not really. Good catch. I will fix this.